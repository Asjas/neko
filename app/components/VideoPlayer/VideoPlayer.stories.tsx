import { ComponentStory, ComponentMeta } from "@storybook/react";

import { VideoPlayer } from "./VideoPlayer";

export default {
  title: "Example/VideoPlayer",
  component: VideoPlayer,
} as ComponentMeta<typeof VideoPlayer>;

const Template: ComponentStory<typeof VideoPlayer> = (args) => <VideoPlayer {...args} />;

export const Primary = Template.bind({});

Primary.args = {
  primary: true,
  label: "Button",
};
