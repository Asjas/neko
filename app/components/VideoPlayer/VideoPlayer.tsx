import { useState } from "react";
import { Stream } from "@cloudflare/stream-react";

function VideoPlayer(props: any) {
  const [stop, setStopped] = useState(false);

  const videoIdOrSignedUrl = "616cb12aac6d7baab9ea6869e131ab01";

  function handleOnEnded(event: any) {
    console.log(event);
  }

  return (
    <div>
      <Stream
        controls
        // autoplay
        onPlay={(e: any) => console.log(e)}
        onEnded={(e: any) => console.log(e)}
        preload="auto"
        muted={true}
        loop={true}
        src={videoIdOrSignedUrl}
      />
    </div>
  );
}

export default VideoPlayer;
