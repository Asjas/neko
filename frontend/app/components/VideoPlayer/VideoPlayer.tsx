import ReactPlayer from "react-player";

export default function VideoPlayer() {
  return (
    <div className="relative aspect-w-16 aspect-h-9">
      <ReactPlayer
        url="https://streamable.com/vuf9ox"
        className="absolute top-0 left-0"
        config={{ file: { attributes: { disablepictureinpicture: "true" } } }}
        width="100%"
        height="100%"
        playing={true}
        volume={0}
        pip={false}
        loop={true}
        muted={true}
        controls
      />
    </div>
  );
}
