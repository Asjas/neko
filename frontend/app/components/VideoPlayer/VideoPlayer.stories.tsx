/* eslint-disable react/function-component-definition */
/* eslint-disable import/no-extraneous-dependencies */
import { ComponentStory, ComponentMeta } from "@storybook/react";

import VideoPlayer from "./VideoPlayer";

export default {
  title: "Example/VideoPlayer",
  component: VideoPlayer,
} as ComponentMeta<typeof VideoPlayer>;

const Template: ComponentStory<typeof VideoPlayer> = (args) => <VideoPlayer {...args} />;

export const Primary = Template.bind({});
