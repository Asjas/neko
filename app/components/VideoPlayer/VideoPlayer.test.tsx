import { render, screen } from "@testing-library/react";
import VideoPlayer from ".";

describe("<VideoPlayer />", () => {
  it("renders correctly", () => {
    render(<VideoPlayer />);

    const loadingText = screen.getByText(/Video Player is loading/i);

    expect(loadingText).toBeInTheDocument();
  });

  it("matches snapshot", () => {
    render(<VideoPlayer />);

    expect(screen.getByLabelText("Video Player")).toMatchSnapshot();
  });
});
