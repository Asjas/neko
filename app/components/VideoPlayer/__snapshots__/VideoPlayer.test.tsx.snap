// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<VideoPlayer /> renders and matches snapshot 1`] = `
<div>
  <div
    class="relative aspect-w-16 aspect-h-9"
  >
    <div
      class="absolute top-0 left-0"
      style="width: 100%; height: 100%;"
    />
  </div>
</div>
`;
